extensions/core/interfaces/map/map.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * This mixin has common imports, methods & variables required to display map on 'input' or 'display'
+ */
+import mixin from "../../../mixins/interface";
+import leaflet from "leaflet";
+import "leaflet/dist/leaflet.css";
+
+const mapMixin = {
+  mixins: [mixin],
+  data() {
+    return {
+      map: null,
+      marker: null,
+      latlng: {
+        lat: null,
+        lng: null
+      }
+    };
+  },
+  watch: {
+    //TODO: Check if this watcher is required
+    //? Do we need to re-render map when theme changes from interface debugger?
+    "options.theme": function(newVal) {
+      leaflet.tileLayer(newVal).addTo(this.map);
+    },
+
+    // Automatically update the Marker based on lat & long
+    latlng: function(newVal) {
+      this.setMarker(newVal);
+    }
+  },
+  methods: {
+    createMap(latlng) {
+      this.map = leaflet.map(this.mapPlaceholder, {
+        center: latlng,
+        zoom: this.options.zoom,
+        maxZoom: this.options.maxZoom,
+        zoomControl: false
+      });
+
+      /**
+       * Set tileLayer
+       * tileLayer defines the interface/theme of the map
+       * There are serveral tileLayers available here: http://leaflet-extras.github.io/leaflet-providers/preview/
+       */
+      leaflet
+        .tileLayer(this.options.theme, {
+          attribution:
+            '&copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>'
+        })
+        .addTo(this.map);
+
+      /**
+       * Bind interaction method only in "input" mode
+       * For "display" mode, interactions are not required.
+       */
+      this.setValue(latlng);
+      if (this.mode == "input") this.bindMapInteractions();
+    },
+
+    /**
+     * Handles Marker Positioning
+     */
+    setMarker(latlng) {
+      if (this.marker) {
+        if (latlng) {
+          this.marker.setLatLng(latlng).setOpacity(1);
+        } else {
+          this.marker.setOpacity(0);
+        }
+      } else if (this.map) {
+        // Create A Marker Instance
+        let markerIcon = leaflet.icon({
+          //? Should we use base64?
```

Comment:
What are the alternatives? An `svg` would already be nicer than a `png` 🙂

Reply:
How did I forget SVG?! 😅I've used inline SVG now.